### PR TITLE
Support modifiers

### LINKED ISSUE #84: Modifiers added with `add_modifier` are not called

### PR DESCRIPTION
**Pull request by dtolnay**
Fixes #84, at least the special case of modifiers with exactly 1 output.

cc @mrmonday

### CODE REVIEW

**Comment by dtolnay** [comment]
I confirmed that libpnet `cargo build` succeeds using this commit and made a PR https://github.com/libpnet/libpnet/pull/211.

File: syntex_syntax/src/ext/decorator.rs [hunk 1]
```diff
@@ -181,16 +203,50 @@ fn expand_decorators_3(
...
+                    panic!("expected 1 output from `#[{}]` but got {}",
+                           mname, modified.len());
+                }
+                let modified = modified.pop().unwrap();
+
+                fld.cx.bt_pop();
+
+                let mut expanded = expand_annotatable(modified, fld);
+                if expanded.is_empty() {
+                    panic!("expected 1 output from `#[{}]` but got {}",
```
Comment (oli-obk) [inline syntex_syntax/src/ext/decorator.rs]: that message isn't correct wrt the condition.

File: syntex_syntax/src/ext/decorator.rs [hunk 2]
```diff
@@ -149,23 +160,34 @@ fn expand_decorators_2(
...
+//     struct A;
+fn expand_3(
+    item: Annotatable,
     attr: &ast::Attribute,
     fld: &mut MacroExpander,
-    decorator_items: &mut SmallVector<Annotatable>,
+    out_items: &mut SmallVector<Annotatable>,
     mname: Name,
-) -> bool {
+) -> (bool, Annotatable) {
```
Comment (oli-obk) [inline syntex_syntax/src/ext/decorator.rs]: not really important, but isn't this basically a `Result<Annotatable, Annotatable>`, where the `Ok` means "handled", and the `Err` means "not handled, here's your `Annotatable` back"

Comment (dtolnay) [inline syntex_syntax/src/ext/decorator.rs]: I changed it to

``` rust
enum Expansion {
    Handled(Annotatable),
    NotHandled(Annotatable),
}
```

File: syntex_syntax/src/ext/decorator.rs (hunk 1 above)
Comment (dtolnay) [inline syntex_syntax/src/ext/decorator.rs]: Fixed.

**Comment by oli-obk** [comment]
@homu r+

**Comment by oli-obk** [comment]
:( no homu here

**Pull request merged**